apm-agent-plugins/apm-opentracing-plugin/src/main/java/co/elastic/apm/agent/opentracingimpl/ApmSpanBuilderInstrumentation.java

```diff
@@ -76,29 +76,26 @@ public static Object createSpan(@Advice.Argument(value = 0, typing = Assigner.Ty
                                             @Advice.FieldValue(value = "operationName") String operationName,
                                             @Advice.FieldValue(value = "microseconds") long microseconds,
                                             @Advice.Argument(1) @Nullable Iterable<Map.Entry<String, String>> baggage) {
-                AbstractSpan<?> parent = null;
-                if (parentContext instanceof AbstractSpan<?>) {
-                    parent = (AbstractSpan<?>) parentContext;
+                AbstractDirectSpan<?> parent = null;
+                if (parentContext instanceof AbstractDirectSpan<?>) {
+                    parent = (AbstractDirectSpan<?>) parentContext;
                 }
                 return doCreateTransactionOrSpan(parent, tags, operationName, microseconds, baggage, PrivilegedActionUtils.getClassLoader(spanBuilderClass));
             }
 
             @Nullable
-            public static AbstractSpan<?> doCreateTransactionOrSpan(@Nullable AbstractSpan<?> parentContext,
+            public static AbstractSpan<?> doCreateTransactionOrSpan(@Nullable AbstractDirectSpan<?> parentContext,
                                                                     Map<String, Object> tags,
                                                                     String operationName, long microseconds,
                                                                     @Nullable Iterable<Map.Entry<String, String>> baggage, ClassLoader applicationClassLoader) {
-                AbstractSpan<?> result = null;
-                ElasticApmTracer tracer = OpenTracingBridgeInstrumentation.tracer.require(ElasticApmTracer.class);
```

Comment:
[question] does it means that we get rid of the OpenTracing bridge ? We are thinking about removing that for a while, but doing so would also require us to officially deprecate and document this.

Reply:
Probably, but I would separate this out into another PR.